pbm/util/errgroup.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"runtime"
+	"sync"
+)
+
+type errorGroup struct {
+	errs []error
+	mu   sync.Mutex
+
+	wg  sync.WaitGroup
+	sem chan struct{}
```

Comment:
This is nice semaphore implementation:
https://pkg.go.dev/golang.org/x/sync@v0.8.0/semaphore